sklearn/tests/test_calibration.py

```diff
@@ -293,15 +286,19 @@ def test_calibration_prob_sum():
     probs = clf_prob.predict_proba(X)
     assert_array_almost_equal(probs.sum(axis=1), np.ones(probs.shape[0]))
 
+
+def test_calibration_less_classes():
     # Test to check calibration works fine when train set in a test-train
     # split does not contain all classes
     # Since this test uses LOO, at each iteration train set will not contain a
     # class label
     X = np.random.randn(10, 5)
     y = np.arange(10)
     clf = LinearSVC(C=1.0)
-    clf_prob = CalibratedClassifierCV(clf, method="sigmoid", cv=LeaveOneOut())
-    clf_prob.fit(X, y)
-    probs = clf_prob.predict_proba(X)
-    n_classes = len(y)
-    assert_array_almost_equal(probs, 1/n_classes)
+    cal_clf = CalibratedClassifierCV(clf, method="sigmoid", cv=LeaveOneOut())
+    cal_clf.fit(X, y)
+
+    for i, calibrated_classifier in \
+            enumerate(cal_clf.calibrated_classifiers_):
+        assert_array_equal(calibrated_classifier.predict_proba(X)[:, i],
```

Comment:
Let's make this slightly stronger by checking that this column and only this column is zero:
```python
        proba = calibrated_classifier.predict_proba(X)
        assert_array_equal(proba[:, i], np.zeros(len(y)))
        assert_equal(np.all(np.hstack([proba[:, :i], proba[:, i+1:]])), True)
```